libflux/src/core/ast/mod.rs

```diff
@@ -537,6 +538,49 @@ pub struct ArrayType {
     pub monotype: MonoType,
 }
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct FunctionType {
+    #[serde(skip_serializing_if = "BaseNode::is_empty")]
+    #[serde(default)]
+    #[serde(flatten)]
+    pub base: BaseNode,
+    pub parameters: Option<Vec<ParameterType>>,
```

Comment:
This just hit me right now, but wrapping this vector in an option is redundant. If the function doesn't take any input then this can just be an empty (length zero) vector.